nwb_conversion_tools/json_schema_utils.py

```diff
@@ -53,33 +53,41 @@ def get_schema_from_method_signature(class_method, exclude=None):
 
     """
     if exclude is None:
-        exclude = []
+        exclude = ['self']
+    else:
+        exclude = exclude + ['self']
     input_schema = get_base_schema()
-    annotation_json_type_map = dict(
-        bool="boolean",
-        str="string",
-        int="number",
-        float="number",
-        dict="object",
-        list="array"
-    )
-
-    for param in inspect.signature(class_method).parameters.values():
-        if param.name not in exclude + ['self']:
+    annotation_json_type_map = {
+        bool: "boolean",
+        str: "string",
+        int: "number",
+        float: "number",
+        dict: "object",
+        list: "array"
+    }
+
+    for param_name, param in inspect.signature(class_method).parameters.items():
+        if param_name not in exclude:
             if param.annotation:
-                param_type = annotation_json_type_map[str(param.annotation).split("'")[1]]
+                param_type = [
+                    annotation_json_type_map[arg_type] for arg_type in param.annotation.__args__
+                    if arg_type in annotation_json_type_map
+                ]
+                assert len(param_type) == 1, \
+                    f"There must be only one valid annotation type that maps to json! {len(param_type)} found."
```

Comment:
```suggestion
                if any(np.isin(param.annotation.__args__, annotation_json_type_map):
                    param_type = annotation_json_type_map[arg_type]
                else:
                    raiseValueError(f"There must be only one valid annotation type that maps to json! {len(param_type)} found.")
```

what about this?

Reply:
Not working out of the box. Here's a simple one liner for testing
`
np.isin((str,pathlib.Path), {str: "string"})
`

which is strange because
`
(str,pathlib.Path)[0] in {str: "string"}
`
works fine.

I'm thinking maybe numpy does some casting that interferes with the class types somehow...

Reply:
Could always replace the `np.isin` with
`
[x in annotation_json_type_map for x in param.annotation.__args__]
`
to the same effect, though. Would that work for you?

Reply:
how about now?

Reply:
[x in annotation_json_type_map for x in param.annotation.__args__]

yeah, that would work!

Reply:
Sadly, no. I had tried that as well, but pulling out the keys specifically doesn't change anything in the `np` version... Very, very strange.

I'll just use the manual compression then to try to get this down a couple more lines of code.